### PR TITLE
cython scons builder

### PR DESCRIPTION
**Pull request by grekiki**
Add custom cython builder

### CODE REVIEW

File: messaging/__init__.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # must be build with scons
 from .messaging_pyx import Context, Poller, SubSocket, PubSocket  # pylint: disable=no-name-in-module, import-error
 from .messaging_pyx import MultiplePublishersError, MessagingError  # pylint: disable=no-name-in-module, import-error
-import capnp
+import capnp
```
Comment (adeebshihadeh) [inline messaging/__init__.py:4]: revert this

File: SConstruct [hunk 2]
```diff
@@ -3,10 +3,12 @@ import distutils
...
 import subprocess
 import sys
+from sysconfig import get_paths
 
 zmq = 'zmq'
 arch = subprocess.check_output(["uname", "-m"], encoding='utf8').rstrip()
 
+python_path = get_paths()['include']
 # Rebuild cython extensions if python, distutils, or cython change
 cython_dependencies = [Value(v) for v in (sys.version, distutils.__version__, Cython.__version__)]
```
Comment (adeebshihadeh) [inline SConstruct:13]: this can be deleted now

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -24,7 +24,7 @@ repos:
         language: system
         types: [python]
         args:
-        - --disable=C,R,W0613,W0511,W0212,W0201,W0311,W0106,W0603,W0621,W0703,E1136
+        - --disable=C,R,W0613,W0511,W0212,W0201,W0311,W0106,W0603,W0621,W0703,E1136,E0401,I1101
```
[the remark this answers is not in the document]
Comment (grekiki) [reply .pre-commit-config.yaml:27]: https://github.com/commaai/cereal/runs/1396170452

Looks like an unimortant issue

File: SConstruct (hunk 2 above)
Comment (grekiki) [reply SConstruct:13]: I still need to link with python.h to run .cpp files generated by cythonize

Comment (adeebshihadeh) [reply SConstruct:13]: I meant line 13

Comment (grekiki) [reply SConstruct:13]: Oh this is just for custom builders? I guess I can also remove it in openpilot then

**Pull request merged**